pkg/storageincentives/agent.go

```diff
@@ -302,6 +307,60 @@ func (a *Agent) start(blockTime time.Duration, blocksPerRound, blocksPerPhase ui
 	}
 }
 
+func purgeStaleDataHandler(logger log.Logger, store storage.StateStorer, currentRound uint64) {
+	if currentRound <= purgeStaleDataThreshold {
+		return
+	}
+
+	thresholdRound := currentRound - purgeStaleDataThreshold
+	keysForRemoval := make([]string, 0)
+
+	stripRoundValue := func(key string) (uint64, error) {
```

Comment:
I have realized the same when I was refactoring agent in later PRs. 

Since this logic for storing round data was already merged, and potentially used by some nodes I think we should merge this PR anyway - just to have complete round data logic in place. Later we could improve this, by storing round data via state, and then remove mechanism for storing round data manually (current approach).

Reply:
> potentially used by some nodes
 
 the release is not out yet so we are still in the green to make the fixes 👍